sdmetrics/single_table/detection/base.py

```diff
@@ -76,6 +76,23 @@ def compute(cls, real_data, synthetic_data, metadata=None):
             transformed_real_data = real_data
             transformed_synthetic_data = synthetic_data
 
+        if metadata is not None and 'columns' in metadata:
+            drop_columns = []
+            for column in metadata['columns']:
+                if 'primary_key' in metadata and column == metadata['primary_key']:
```

Comment:
why don't these get dropped?

Reply:
Cleaned logic, now it shouldn't need to skip primary keys